### PR TITLE
Se 592 update solidity compiler version

### PR DESCRIPTION
**Pull request by mpetersoCode55**
Updated solidity, foundry and OZ dependencies to use the latest versions of each.

### CODE REVIEW

**Comment by ShaneDuncan602** [comment]
We also need to change the main README to reflect the changes. It still has 4.9 for the versions and talks about npm install. That should have forge install with the proper version.

Also, it looks like there are also still dependencies in the github workflows.

File: script/python/get_selectors.py [hunk 1]
```diff
@@ -8,11 +8,12 @@
...
-    res = subprocess.run(
-        ["forge", "inspect", contract, "mi"], capture_output=True)
-    res = res.stdout.decode()
-    res = json.loads(res)
+    line = ""
+    while line == "":
+        res = subprocess.run(
+            ["forge", "inspect", contract, "mi"], capture_output=True )
+        line = res.stdout.decode()
+    res = json.loads(line)
```
Comment (ShaneDuncan602) [inline script/python/get_selectors.py:16]: This was an awesome change.

**Review by ShaneDuncan602 — CHANGES_REQUESTED** [review]
Found a few missed dependencies but this is fantastic. Great work.

File: docs/userGuides/deployment/DEPLOY-PROTOCOL.md [hunk 2]
```diff
@@ -35,13 +35,6 @@ is an overview of this deployment process:
    foundryUp
    ````
 
-   #### *Troubleshooting*
-
-   *if deployment fails (following the next steps), it might be caused by a faulty current version of Foundry. Try installing this specific version instead:*
-
-   ```bash
-   foundryup --version nightly-09fe3e041369a816365a020f715ad6f94dbce9f2
-   ```
```
Comment (oscarsernarosero) [inline docs/userGuides/deployment/DEPLOY-PROTOCOL.md:44]: great catch

File: package.json [hunk 3]
```diff
@@ -7,8 +7,8 @@
   "version": "1.1.0",
   "description": "Set of contracts to easily start an application in rules protocol",
   "dependencies": {
-    "@openzeppelin/contracts": "4.9",
-    "@openzeppelin/contracts-upgradeable": "4.9"
+    "@openzeppelin/contracts": "5.0.2",
+    "@openzeppelin/contracts-upgradeable": "5.0.2"
```
Comment (oscarsernarosero) [inline package.json:11]: nice

**Review by oscarsernarosero — APPROVED** [review]
Looks great Michael. Im glad it didn't require lots of detail fixing.

**Pull request merged**